hyperspy/tests/axes/test_data_axis.py

```diff
@@ -314,6 +314,46 @@ def test_calibrate(self):
         with pytest.raises(TypeError, match="only for uniform axes"):
             self.axis.calibrate(value_tuple=(11,12), index_tuple=(0,5))
 
+    def test_functional_value2index(self):
```

Comment:
For the other two axes types, the different types of inputs for `value2index` and possible errors are tested in a number of individual functions, while here they are all in one function. Wouldn't it make sense to decide for one way for all three axes types? I'm fine with either.

Reply:
Thinking about it, the single function is probably more clear. Would you want to adapt it for the other axes types?

Reply:
Should be good now